README
NAME
    Pod::Usage::Return - pod2usage that returns instead of exits

VERSION
    version 0.002

SYNOPSIS
        use Pod::Usage::Return;
        use Pod::Usage::Return qw( pod2usage );

        exit pod2usage(0);

        sub main {
            return pod2usage("ERROR: An error occurred!") if $ERROR;
        }

        exit pod2usage( -exitval => 1, -message => 'ERROR: An error occurred' );

DESCRIPTION
    This is a drop-in replacement for Pod::Usage "pod2usage" that returns
    the exit value instead of calling exit.g

FUNCTIONS
  pod2usage
    See Pod::Usage for documentation. Returns the exit code instead of
    calling exit().

AUTHOR
    Doug Bell <preaction@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Doug Bell.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



CHANGES
==================================================
Changes from 2013-05-10 00:00:00 +0000 to present.
==================================================

------------------------------------------
version 0.002 at 2014-05-10 02:39:51 +0000
------------------------------------------

  Change: 1cfe3bbe108e5e22fe6a2700de740f575b9bd15e
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-09 21:39:43 +0000

    update build artifacts 

  Change: b7f5acadb82ed8148e29339ae60f9e9719c345ab
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-09 21:39:33 +0000

    tests require Capture::Tiny 

  Change: d82e10f38c73209770626d5d15f1bbef5c942a5a
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-09 21:38:54 +0000

    update build artifacts 

  Change: 6c3b5a8ae210631ac3bca3a064261e52ba76e9db
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-09 21:38:20 +0000

    add a minimum version for Pod::Usage

    This is the least version I could find, and it still works. 

------------------------------------------
version 0.001 at 2014-05-09 04:05:38 +0000
------------------------------------------

  Change: 9c44ef271ae8e2ba5204c241cbceb08e2c6da802
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-08 23:05:38 +0000

    initial build artifacts 

  Change: 1731dccd8dc18ca3dd24921a1bb20a64f241e00a
  Author: Doug Bell <madcityzen@gmail.com>
  Date : 2014-05-08 23:00:48 +0000

    initial pod2usage use-cases

    I only use the pod2usage( EXITCODE ) and pod2usage( MESSAGE ), really 

================
End of releases.
================


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "pod2usage that returns instead of exits",
  "dist_author" => [
    "Doug Bell <preaction\@cpan.org>"
  ],
  "dist_name" => "Pod-Usage-Return",
  "dist_version" => "0.002",
  "license" => "perl",
  "module_name" => "Pod::Usage::Return",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Pod::Usage" => "1.36"
  },
  "script_files" => [],
  "test_requires" => {
    "Capture::Tiny" => 0,
    "Test::Compile" => 0,
    "Test::Most" => 0
  }
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Pod/Usage/Return.pm
package Pod::Usage::Return;
{
  $Pod::Usage::Return::VERSION = '0.002';
}
# ABSTRACT: pod2usage that returns instead of exits

use strict;
use warnings;
use Pod::Usage ();
use base 'Exporter';
our @EXPORT = qw( pod2usage );

sub pod2usage {
    my $exitval = 2;
    my %args;
    if ( @_ == 1 ) {
        if ( ref $_[0] eq 'HASH' ) {
            %args = %{$_[0]};
        }
        elsif ( $_[0] =~ /^\d+$/ ) {
            $exitval = $_[0];
        }
        else {
            $args{-message} = $_[0];
            $args{-verbose} = 0;
        }
    }
    else {
        %args = @_;
    }
    $args{-exitval} = 'NOEXIT';
    if ( $exitval >= 2 ) {
        $args{-output} = *STDERR;
    }
    Pod::Usage::pod2usage( \%args );
    return $exitval;
}

1;

__END__

=pod

=head1 NAME

Pod::Usage::Return - pod2usage that returns instead of exits

=head1 VERSION

version 0.002

=head1 SYNOPSIS

    use Pod::Usage::Return;
    use Pod::Usage::Return qw( pod2usage );

    exit pod2usage(0);

    sub main {
        return pod2usage("ERROR: An error occurred!") if $ERROR;
    }

    exit pod2usage( -exitval => 1, -message => 'ERROR: An error occurred' );

=head1 DESCRIPTION

This is a drop-in replacement for L<Pod::Usage> C<pod2usage> that returns the
exit value instead of calling exit.g

=head1 FUNCTIONS

=head2 pod2usage

See L<Pod::Usage> for documentation. Returns the exit code instead of calling
exit().

=head1 AUTHOR

Doug Bell <preaction@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Doug Bell.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "pod2usage that returns instead of exits",
  "AUTHOR" => "Doug Bell <preaction\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Pod-Usage-Return",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Pod::Usage::Return",
  "PREREQ_PM" => {
    "Pod::Usage" => "1.36"
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Test::Compile" => 0,
    "Test::Most" => 0
  },
  "VERSION" => "0.002",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



